core/repository/transaction.ts

```diff
@@ -84,20 +97,21 @@ class TransactionRepo implements ITransactionRepository<object> {
 
     dbSave(trs: Transaction<object>): IDBTransactionSave { return undefined; }
 
-    public saveTransaction = async (transaction): Promise<Response<void>> => {
-        this.memoryTransactions[transaction.id] = transaction;
-
-        try {
-            await db.tx(async (t) => {
-                const promise = await this.dbSave(transaction);
-                const inserts = new Inserts(promise, promise.values);
-                const promises = t.none(inserts.template(), promise.values);
-                await t.batch(promises);
-            });
-        } catch (error) {
-            logger.error(`[Chain][saveTransaction][tx]: ${error}`);
-            logger.error(`[Chain][saveTransaction][tx][stack]:\n${error.stack}`);
-        }
+    public saveTransaction = async (transaction: Transaction<object>): Promise<Response<void>> => {
+        // try {
+        //     await db.tx(async (t) => {
+        //         const promise = await this.dbSave(transaction);
+        //         const inserts = new Inserts(promise, promise.values);
+        //         const promises = t.none(inserts.template(), promise.values);
+        //         await t.batch(promises);
+        //     });
+        // } catch (error) {
+        //     logger.error(`[Chain][saveTransaction][tx]: ${error}`);
+        //     logger.error(`[Chain][saveTransaction][tx][stack]:\n${error.stack}`);
+        // }
+        // return new Response<void>();
+        this.memoryTransactionByBlockId[transaction.blockId].push(transaction);
```

Comment:
this.memoryTransactionByBlockId[transaction.blockId] can return undefined and you need to initialize empty array